### ISSUE TITLE
Multiple select redesign

### ISSUE THREAD

**Issue by OzrenMuic** [huridocs/uwazi#2609]
### Why change
- Because in some cases multiple select of large thesauri takes a lot of space (in some cases even few screens - like _in_ _'Country'_ filter on _Atlas of Torture_.
- Also, making multiple select small will help us later when restructuring navigation. 
- This design will conserve space and add to usability.
- Allow us to prominently suggest labels with topic classification.

### Design
<img width="553" alt="Screenshot 2019-10-24 at 22 57 52" src="https://user-images.githubusercontent.com/8134850/67524853-d5e96d80-f6b1-11e9-9e9e-9659446f81bc.png">
Link to design: 
https://www.figma.com/file/xjsz7r65epN4B0b96zilac/multiple-select?node-id=0%3A1

You can find similar behaviour here:
 https://semantic-ui.com/modules/dropdown.html#multiple-search-selection

**Comment by txau** [comment]
@OzrenMuic I agree with the proposal.

Today we were having a conversation related to this since there is a negative impact in performance coming from multi-selects, and maybe this needs to be factored in the design.

The problem happens when we have too many options in the multi-select, since they can be used for a relationship field, so eventually you end up with thousands of values. The proposed tech solutions are:
- Pagination in the values (Show more link at the bottom loading batches of N values).
- Only be able to access all the options by running a search, so you will never see the whole list.

The performance issue also happens for selects (not multi). I was wondering if at the end of the day a regular select is the same as a multi-select restricted to only one value?

cc: @RafaPolit @konzz @daneryl

**Comment by txau** [comment]
BTW I guess in the figma, the screen "new open + selecting" still shows the "+ Select options" box. From the states design I assume this is a mistake?

**Comment by RafaPolit** [comment]
As mentioned, we were just discussing this.  It’s a great improvement to our current look.

What are the numbers at the far right of the open options?  Is that an aggregation of how many other fields have that value?  I so, I assume that is for this particular template and property type?

I would not introduce that value at this point.  It could be misleading depending on template naming conventions and other search-related factors.

**Comment by RafaPolit** [comment]
Also, some of those options can have considerably longer names, some amounting to a full sentence.  How would that show? Would you have a multi-line tag? Will it ellipsis the result?

**Comment by OzrenMuic** [comment]
> BTW I guess in the figma, the screen "new open + selecting" still shows the "+ Select options" box. From the states design I assume this is a mistake?

Yes. Edited it. 👍

**Comment by OzrenMuic** [comment]
> The problem happens when we have too many options in the multi-select, since they can be used for a relationship field, so eventually you end up with thousands of values. 

Is there example that I can see where this happens?

> I was wondering if at the end of the day a regular select is the same as a multi-select restricted to only one value?

I would say yes. 
In my design only difference would be that it stays open until you finish selecting.

**Comment by OzrenMuic** [comment]
> As mentioned, we were just discussing this. It’s a great improvement to our current look.

Thanks guys!

> What are the numbers at the far right of the open options? Is that an aggregation of how many other fields have that value? I so, I assume that is for this particular template and property type?

It's number of entities with this label. These aggregations are already there for multiple select.

![image](https://user-images.githubusercontent.com/8134850/67530327-17cce080-f6bf-11e9-9a1f-1bc16e475289.png)

**Comment by OzrenMuic** [comment]
> Also, some of those options can have considerably longer names, some amounting to a full sentence. How would that show? Would you have a multi-line tag? Will it ellipsis the result?

That is a problem. I wouldn't go into two lines. 64 char should be enough for label. We should write best practices for labelling, see who has too long labels and nudge them into considering shortening them with a message that we'll be cutting off labels over 64 char.
Just my opinion.

![image](https://user-images.githubusercontent.com/8134850/67530523-99247300-f6bf-11e9-8644-dd179502323c.png)

**Comment by txau** [comment]
Regarding the long labels, while I think as a general idea they should be short, keep in mind that often these values are coming from official naming of documents or institutions and can't be shortened.

Ie. Check the filters for violated articles here: https://summa.cejil.org/en/library/?q=(_types:!(%2758b2f3a35d59f31e1345b48a%27),filters:(),limit:30,order:desc,sort:creationDate,types:!(%2758b2f3a35d59f31e1345b48a%27))

![Screenshot from 2019-10-25 05-54-22](https://user-images.githubusercontent.com/489831/67569396-fd5d2c00-f6eb-11e9-9302-0a5b3d0dbe55.png)

This screenshot points me to another question, how is nesting going to work in this design?

Regarding the multi-selects with too many options, here is an example also from SUMMA. When editing a process, since the documents are connected and displayed using a relationship field, they look like this when editing:

![Screenshot from 2019-10-25 05-56-51](https://user-images.githubusercontent.com/489831/67569552-6a70c180-f6ec-11e9-969d-3b02d94d389b.png)

**Comment by OzrenMuic** [comment]
I think expecting that system will morph into anything you want is unrealistic. What if titles are 500 words long? Would admin still display it that way? I bet they would find a way to shorten it so it's more usable to the user. I would go so far to display two lines before truncating it and display it in full on hover. (Example on the left).

But I didn't find in CEJIL long items, only long group titles. We could maybe display items one after another so it looks like right example in CEJIL's case.

![image](https://user-images.githubusercontent.com/8134850/67677915-9195f000-f985-11e9-807b-25201950c635.png)

#### Nested items
I wouldn't recommend nested items. I would just put lists of titles. 
Search would filter both titles and items. If search is catching a title, then we show all children. Otherwise show only children with title over it. (example below)

![image](https://user-images.githubusercontent.com/8134850/67679209-df602780-f988-11e9-9c31-ab5b9af22599.png)

I feel that there's a lot of added complexity blocks users from using some features. I think that we need to simplify the whole thing.

Let me know what you think...

**Comment by OzrenMuic** [comment]
> Regarding the multi-selects with too many options, here is an example also from SUMMA. 

Wow.. That is bad. I'm speechless.

**Comment by txau** [comment]
@OzrenMuic I agree with you in having a middle ground regarding long titles. Two lines and then truncating should be more than enough!

Regarding the nesting. We call it nesting but actually the behavior is what you are describing, its groups of items with a title. The are some particularities to this approach such as, users can click on the title and it will select all the nested items.

**Comment by OzrenMuic** [comment]
Ok, I'll add expanding titles

**Comment by OzrenMuic** [comment]
> Regarding the long labels, while I think as a general idea they should be short, keep in mind that often these values are coming from official naming of documents or institutions and can't be shortened.
> 
> Ie. Check the filters for violated articles here: https://summa.cejil.org/en/library/?q=(_types:!(%2758b2f3a35d59f31e1345b48a%27),filters:(),limit:30,order:desc,sort:creationDate,types:!(%2758b2f3a35d59f31e1345b48a%27))
> 
> ![Screenshot from 2019-10-25 05-54-22](https://user-images.githubusercontent.com/489831/67569396-fd5d2c00-f6eb-11e9-9302-0a5b3d0dbe55.png)

BTW About the above example, we should not take it into account. PM on that project should contact them and suggest to write full titles. 

![image](https://user-images.githubusercontent.com/8134850/67885923-c00ff880-fb48-11e9-8a09-5df52e51a264.png)

**Comment by OzrenMuic** [comment]
I've reviewed everything and took in consideration comments here. Here are the updates

prototype: https://www.figma.com/proto/xjsz7r65epN4B0b96zilac/multiple-select?node-id=42%3A501&viewport=2041%2C-680%2C0.3885210156440735&scaling=min-zoom

design: https://www.figma.com/file/xjsz7r65epN4B0b96zilac/multiple-select?node-id=42%3A1208

**Comment by txau** [comment]
@OzrenMuic currently the multi-select is used both for editing properties and filter the collection. The behavior is pretty much the same. What I'm concerned is that in the use case of filtering the lib, our current approach shows users not only the title of the property, but also the top categories, giving users a lead towards what kind of info they are going to find in the filter. I guess we are loosing that angle. For you to consider.

**Comment by OzrenMuic** [comment]
I understand that is used as navigational affordance but that is unintended use of filter. This should be solved in different way.

I have an idea of adding groupings by metadata like template or some other that would give better overview. This will be very much needed in table view when come to it.

**Comment by txau** [comment]
@OzrenMuic what do you mean by "groupings by metadata like template or some other"?

Still, we need to solve the faceted browsing issue somehow. Maybe its a variation of this same component that shows the top options when folded AND being used as a filter.

**Comment by txau** [comment]
Related: https://github.com/huridocs/uwazi/issues/1917 https://github.com/huridocs/uwazi/issues/2785

**Comment by MatelSow** [comment]
@simonfossom a partner brought up a point which relates to the display of multi-select items but in the full view (not the library view). Would these design options also take into account the display of long multi-select options in the full view?

See tags listed below. Partner asked about a different display of the tags away from the vertical line. 
![1](https://user-images.githubusercontent.com/56391759/79198014-12ad4a80-7e33-11ea-9a09-c04d563d544a.png)

**Comment by simonfossom** [comment]
@MatelSow This is how multiple-select would show many options by the proposed design:

![image](https://user-images.githubusercontent.com/8134850/79236177-eadad880-7e6c-11ea-9f09-22ea699b25b1.png)

I think this would solve it, I think.

**Comment by MatelSow** [comment]
@simonfossom yes this would solve it. Thank you!
